### PR TITLE
App: fix child windows not inheriting mainWindow properties, including userAgent

### PR DESCRIPTION
**Pull request by TheCleric**
When a new child window is spawned (such as for a Google login popup), those child windows were not receiving the mainWindow's properties. Chiefly among this was the userAgent which caused a bug in #831

### CODE REVIEW

File: app/src/main.ts [hunk 1]
```diff
@@ -344,3 +352,35 @@ app.on('login', (event, webContents, request, authInfo, callback) => {
...
+app.on('browser-window-created', (event: Event, window: BrowserWindow) => {
+  log.debug('app.browser-window-created', { event, window });
+  if (setupWindow !== undefined) {
+    setupWindow(window);
+  }
+});
+
+app.on('browser-window-focus', (event: Event, window: BrowserWindow) => {
+  log.debug('app.browser-window-focus', { event, window });
+});
```
Comment (ronjouch) [inline app/src/main.ts:386]: Why subscribe to these events (except `'browser-window-created'`) and not do anything with them?

**Review by ronjouch — APPROVED** [review]
👍, only one question. I'll read your reply on why you may want, or not, to insist on handling all these events, but regardless, good, feel free to merge.

**Comment by TheCleric** [comment]
> 👍, only one question. I'll read your reply on why you may want, or not, to insist on handling all these events, but regardless, good, feel free to merge.

@ronjouch I actually started adding all events to see what was being fired off to see what I could intercept, but got to the one I needed and stopped. I still think it's useful to know what's going on / order of events in a debug scenario and would propose that we add all events like this at some point. 

Also based on the hacky way I had to send that setup func back I think this area is ready for a refractor anyway. What do you think?

**Comment by ronjouch** [comment]
@TheCleric 

> I started adding all events to see what was being fired off to see what I could intercept, but got to the one I needed and stopped. I still think it's useful to know what's going on / order of events in a debug scenario and would propose that we add all events like this at some point.

Makes sense! This costs basically zero, and given the new loglevelled logging, it's invisible by default. Okay with the current code that I'll merge now (to cut a release), and okay to add all events at some point to improve hackability.


> Also based on the hacky way I had to send that setup func back I think this area is ready for a refactor anyway. What do you think?

👍, do all the refactor that's needed to keep things understandable & maintainable.

**Pull request merged**